csaf/src/verification/check/base/mod.rs

```diff
@@ -0,0 +1,128 @@
+use csaf::Csaf;
+// use csaf::document::{Category, CsafVersion, PublisherCategory};
+use crate::verification::check::{Check, CheckError, Checking};
+
+pub fn check_csaf_base(csaf: &Csaf) -> Vec<CheckError> {
+    let mut result = vec![];
+    result.extend(
+        Checking::new()
+            .require(
+                "The csaf file does not have document publisher name",
+                !csaf.document.publisher.name.is_empty(),
+            )
+            .done(),
+    );
+    result.extend(
+        Checking::new()
+            .require(
+                "The csaf file does not have document title",
+                !csaf.document.title.is_empty(),
+            )
+            .done(),
+    );
+    result.extend(
+        Checking::new()
+            .require(
+                "The csaf file's document tracking id is empty",
+                !csaf.document.tracking.id.is_empty(),
+            )
+            .done(),
+    );
+    result
+}
+
+pub fn check_csaf_document_tracking_revision_history(csaf: &Csaf) -> Vec<CheckError> {
+    let mut result = vec![];
+    for revision in &csaf.document.tracking.revision_history {
+        result.extend(
+            Checking::new()
+                .require(
+                    format!(
+                        "The csaf file's document revision_history {:?} number is empty",
+                        revision.number.to_string()
+                    ),
+                    !revision.number.is_empty(),
+                )
+                .done(),
+        );
+        result.extend(
+            Checking::new()
+                .require(
+                    "The csaf file's document revision_history summary is empty",
+                    !revision.summary.is_empty(),
+                )
+                .done(),
+        );
+    }
+    result
+}
+
+pub fn init_csaf_base_verifying_visitor() -> Vec<(&'static str, Box<dyn Check>)> {
+    vec![
+        ("check_csaf_base", Box::new(check_csaf_base)),
+        (
+            "check_csaf_document_tracking_revision_history",
+            Box::new(check_csaf_document_tracking_revision_history),
+        ),
+    ]
+}
+// pub fn check_csaf_publisher_category(csaf: &Csaf) -> Vec<CheckError> {
```

Comment:
I think these functions do not needs check, because  CSAF parse will verify them.
If it has any invalid vaule, will get parse error.